docs/developer-docs/backend/rust/timers.md

```diff
@@ -0,0 +1,278 @@
+* `"counter": () -> (nat64) query` &mdash; declares a canister query entry point named `counter`. The `counter` query takes no arguments `()` and returns an integer `(nat64)`.
+
+The interface definition is complete, save the changes.
+
+Step 4. Implementing the `counter` query
+----------------------------------------
+
+In the Step 3 above, the `counter` query was declared: `"counter": () -> (nat64) query` This step implements it.
+
+In the code editor, open the `src/my_timers_backend/src/lib.rs` file and replace its content with the following:
+
+```rust
+use std::sync::atomic::{AtomicU64, Ordering};
+
+static COUNTER: AtomicU64 = AtomicU64::new(0);
+
+#[ic_cdk_macros::query]
+fn counter() -> u64 {
+    COUNTER.load(Ordering::Relaxed)
+}
+```
+
+Where:
+
+* `static COUNTER: AtomicU64 = ...` &mdash; defines a new global variable called `COUNTER`.
+* `#[ic_cdk_macros::query]` &mdash; marks the following `counter` function as a `query` entry point.
+* `fn counter() -> u64 {...}` &mdash; defines the query. Just like in the `.did` definition, it takes no arguments and returns `u64`.
+* `COUNTER.load(...)` &mdash; load and return the global `COUNTER` value.
+
+Step 5. Implementing canister initialization
+--------------------------------------------
+
+In the Step 3 above, the service declaration has a single integer argument: `service : (nat64) -> {...}` This step implements the canister initialization with the argument.
+
+In the code editor, open the `src/my_timers_backend/src/lib.rs` file and append the following:
+
+```rust
+[...]
+
+#[ic_cdk_macros::init]
+fn init(timer_interval_secs: u64) {
+    let interval = std::time::Duration::from_secs(timer_interval_secs);
+    ic_cdk::println!("Starting a periodic task with interval {interval:?}");
+    ic_cdk_timers::set_timer_interval(interval, || {
+        COUNTER.fetch_add(1, Ordering::Relaxed);
+    });
+}
+```
+
+Where:
+
+* `#[ic_cdk_macros::init]` &mdash; marks the following `init` function as a canister initialization method.
+* `fn init(interval: u64) {...}` &mdash; defines the initialization method. Just like in the `.did` definition, the function takes one argument: timer interval in seconds.
+* `ic_cdk::println!(...)` &mdash; prints the debug log message on the local `dfx` console.
+* `ic_cdk_timers::set_timer_interval(...)` &mdash; creates a new periodic timer with the specified interval and a closure to call.
+* `COUNTER.fetch_add(1, ...)` &mdash; every time the periodic task is triggered, the global `COUNTER` is increased.
+
+Step 6. Implementing canister upgrade
+-------------------------------------
+
+Note, as described in [Periodic Tasks and Timers](/developer-docs/backend/periodic-tasks.md), the timers library does not handle canister upgrades. It is up to the canister developer to serialize the timers in the `canister_pre_upgrade` and reactivate the timers in the `canister_post_upgrade` method if needed.
```

Comment:
link is broken. Preferred way to write this is `../periodic-tasks.md`

Reply:
There are 70 links `(/developer-docs/...`, shall we fix them all (in a separate PR)?